### DISCUSSION TITLE
Is there a way to export to SQL file for Tables/Functions for addition to codebase?

### DISCUSSION THREAD

**Discussion by nruffilo** [supabase/supabase#1266] [Feature Requests]
I maintain my own .sql file with the commands to completely replicate/recreate a project that I'm working on.  This allows all developers (it's an open source project) to create their own supabase instance to test locally and work on their own dev/local version.  But, it would be much easier if I could just run a command and export all my table structures (no data) and functions so that I can guarantee that what I have is the latest and working.  Is that possible?  Additionally, is there any way to link a github repo to a database instance so that it keeps the structure in the code repo?

Thank you!

**Comment by silentworks** [comment]
You can either use a migration framework to do some of this work for you or the Supabase cli now includes a diff tool that will create a migration file for you. The Supabase database information can be found under `settings/database`.

Do not commit this database information into your repo, make sure you store it in a `.env` file or something and add it to the `.gitignore` file.

**Reply by MildTomato** [reply]
Maybe we run that same diff in the dashboard client (provided the org member is admin or owner) and then download SQL file.

**Reply by Fideltodayy** [reply]
Hey man, how you go about creating the migration .sql file from supabase?
